lib/tdf3/src/tdf.ts

```diff
@@ -1061,7 +1061,8 @@ export async function readStream(cfg: DecryptConfiguration) {
 
   const cipher = new AesGcmCipher(cfg.cryptoService);
 
-  await updateChunkQueue(
```

Comment:
Hmm. This means if there is a network error in this function, it won't propagate anymore. Instead of throwing, you will have to explicitly call a _reject handler on the chunk.

- In this method: add a corresponding `_reject` parameter, in addition to the `_resolve` handler below. This then propagate an error in chunk processing to the stream assembly code here by throwing it out on the await on line 1085
```
      if (!chunk.decryptedChunk) {
        await new Promise((resolve, reject) => {
          chunk._resolve = resolve;
          chunk._reject = reject;
        });
      }
```
- Above, in `updateChunkQueue`, instead of throwing the error, send it to the offending slice. It is kinda hard, since you won't know which slice the error happened on in the current double-looped code. So you'll have to refactor it a bit so the handler happens within the innermost loop, and add another handler for the zip format check which just attaches it to the first item in the stride.

Reply:
I think it should be hooked up properly, though not sure exactly which zip format check you meant.